app/models/presenters/tube_presenter.rb

```diff
@@ -60,5 +61,50 @@ def tubes_and_sources
         yield child_tubes if block_given? && child_tubes.present?
       end
     end
+
+    def qc_summary?
+      labware.receptacle&.qc_results&.to_a.present?
+    end
+
+    def qc_summary
+      labware.receptacle.qc_results.sort_by(&:key).each do |result|
+        yield result.key.titleize, result.unit_value.to_s
+      end
+      nil
+    end
+
+    def transfer_volumes?
+      [source_molarity, target_molarity, target_volume, minimum_pick].none?(&:nil?)
+    end
+
+    def source_molarity
+      molarity_qc_result = labware.receptacle.qc_results.detect { |result| result.key == 'molarity' }
```

Comment:
See the `latest_qc` and `latest_molarity` methods on well, which we should probably include on receptacle. Reason being we can have multiple reading taken, and we want to use the most recent reading. This method also double checks the units for us.